src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1847,58 +1920,78 @@ impl Cage {
                             }
 
-                            let newposition;
-                            if let Ok(byteswritten) = fileobject.writeat(buf, count, position) {
-                                //move position forward by the number of bytes we've written
-                                normalfile_filedesc_obj.position = position + byteswritten;
-                                newposition = normalfile_filedesc_obj.position;
-                                if newposition > normalfile_inode_obj.size {
-                                    normalfile_inode_obj.size = newposition;
-                                    drop(inodeobj);
-                                    drop(fileobject);
-                                    log_metadata(&FS_METADATA, normalfile_filedesc_obj.inode);
-                                } //update file size if necessary
-
-                                byteswritten as i32
-                            } else {
-                                0 //0 bytes written, but not an error value
-                                  // that can/should be passed to the user
+                            // Write `count` bytes from `buf` to the file at `position` using
+                            // `writeat` function, which returns the number of bytes written when
+                            // successful, else panics.
+                            let byteswritten = fileobject.writeat(buf, count, position).unwrap();
+                            // Move position forward by the number of bytes we've written
+                            normalfile_filedesc_obj.position = position + byteswritten;
+                            // Update the file size if necessary
+                            if normalfile_filedesc_obj.position > normalfile_inode_obj.size {
+                                normalfile_inode_obj.size = normalfile_filedesc_obj.position;
+                                drop(inodeobj);
+                                drop(fileobject);
+                                log_metadata(&FS_METADATA, normalfile_filedesc_obj.inode);
                             }
+                            // Return the number of bytes written
+                            byteswritten as i32
                         }
 
+                        // For `CharDev` type inode, the writing happens to the Character Device
+                        // file, with each device type returning the `count` number of bytes that
+                        // are to be written.
                         Inode::CharDev(ref char_inode_obj) => {
+                            // The `_write_chr_file` helper function typically does not write
+                            // anything to the device and simply returns the bytes count.
                             self._write_chr_file(&char_inode_obj, buf, count)
                         }
 
+                        // A Sanity check where the File type fd should not have a `Socket` type
```

Comment:
explaining briefly why it shouldn't be a socket would be great

Reply:
Updated.